lib/modules/manager/bundler/extract.ts

```diff
@@ -61,44 +61,69 @@ export async function extractPackageFile(
       dep.datasource = RubyGemsDatasource.id;
       res.deps.push(dep);
     }
-    const groupMatch = regEx(/^group\s+(.*?)\s+do/).exec(line);
-    if (groupMatch) {
-      const depTypes = groupMatch[1]
-        .split(',')
-        .map((group) => group.trim())
-        .map((group) => group.replace(regEx(/^:/), ''));
-      const groupLineNumber = lineNumber;
-      let groupContent = '';
-      let groupLine = '';
-      while (lineNumber < lines.length && groupLine !== 'end') {
-        lineNumber += 1;
-        groupLine = lines[lineNumber];
-        // istanbul ignore if
-        if (!is.string(groupLine)) {
-          logger.debug(
-            { content, packageFile, type: 'groupLine' },
-            'Bundler parsing error',
-          );
-          groupLine = 'end';
+
+    const processGroupBlock = async (
+      line: string,
+      repositoryUrl?: string,
+      trimGroupLine: boolean = false,
+    ): Promise<void> => {
```

Comment:
```suggestion
    async function processGroupBlock(
      line: string,
      repositoryUrl?: string,
      trimGroupLine: boolean = false,
    ): Promise<void> {
```

Reply:
Unfortunatelly linter complains about this suggestion: 
```shell
 65:5  error  Move function declaration to function body root  no-inner-declarations
```

Do you think I can change the function to something like this (as suggested in the exception part of the second bullet point of [Fucntion Best Practices Doc](https://github.com/renovatebot/renovate/blob/main/docs/development/best-practices.md#functions))
```js
const processGroup = async function (
  line: string,
  repositoryUrl?: string,
  trimGroupLine: boolean = false,
): Promise<void> { ... )
```

Reply:
move function to module scope, don't use nested functions (perf)

Reply:
@viceice this function seems to manipulate variables like lineNumber so can't be easily moved

Reply:
Moved to top of function